matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+        String busId = getBusId(mBus.getNumber());
+        LOGGER.debug("Creating bus {}", busId);
+        Bus bus = voltageLevel.getBusBreakerView().newBus()
+                .setId(busId)
+                .setName(busId)
+                .add();
+        bus.setV(mBus.getVoltageMagnitude() * voltageLevel.getNominalV())
+                .setAngle(mBus.getVoltageAngle());
+        return bus;
+    }
+
+    private static Substation createSubstation(Network network, String substationId) {
+        LOGGER.debug("Creating substation {}", substationId);
+        Substation substation = network.getSubstation(substationId);
+        if (substation == null) {
+            substation = network.newSubstation()
+                    .setId(substationId)
+                    .add();
+        }
+        return substation;
+    }
+
+    private static VoltageLevel createVoltageLevel(MBus mBus, String voltageLevelId, Substation substation, Network network, PerUnitContext perUnitContext) {
+        double nominalV = perUnitContext.isIgnoreBaseMva() || mBus.getBaseVoltage() == 0 ? 1 : mBus.getBaseVoltage();
+        VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelId);
+        if (voltageLevel == null) {
+            LOGGER.debug("Creating voltagelevel {}", voltageLevelId);
+            voltageLevel = substation.newVoltageLevel()
+                    .setId(voltageLevelId)
+                    .setNominalV(nominalV)
+                    .setTopologyKind(TopologyKind.BUS_BREAKER)
+                    .add();
+        }
+        return voltageLevel;
+    }
+
+    private static void createLoad(MBus mBus, VoltageLevel voltageLevel) {
+        if (mBus.getRealPowerDemand() != 0 || mBus.getReactivePowerDemand() != 0) {
+            String busId = getBusId(mBus.getNumber());
+            String loadId = busId + "-L";
+            LOGGER.debug("Creating load {}", loadId);
+            voltageLevel.newLoad()
+                .setId(loadId)
+                .setConnectableBus(busId)
+                .setBus(busId)
+                .setP0(mBus.getRealPowerDemand())
+                .setQ0(mBus.getReactivePowerDemand())
+                .add();
+        }
+    }
+
+    private static String createGenId(MGen mGen, Network network) {
+        String genIdPrefix = "G" + mGen.getNumber() + "-";
+        int uniqueGenSuffix = 0;
+        String genId;
+        do {
+            genId = genIdPrefix + uniqueGenSuffix++;
+        } while  (network.getIdentifiable(genId) != null);
+        return genId;
+    }
+
+    private static void createShuntCompensator(MBus mBus, VoltageLevel voltageLevel, PerUnitContext perUnitContext) {
+        if (mBus.getShuntSusceptance() != 0) {
+            String busId = getBusId(mBus.getNumber());
+            String shuntId = busId + "-SH";
+            LOGGER.debug("Creating shunt {}", shuntId);
+            double zb = Math.pow(voltageLevel.getNominalV(), 2) / perUnitContext.getBaseMva();
```

Comment:
I'm not sure Math.pow is really efficient to compute a square... use `voltageLevel.getNominalV() * voltageLevel.getNominalV() / perUnitContext.getBaseMva())`

Reply:
you're right! done.